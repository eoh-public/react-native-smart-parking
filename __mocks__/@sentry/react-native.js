export const setUser = jest.fn();
export const init = jest.fn();

export default {
  init,
  setUser,
};
