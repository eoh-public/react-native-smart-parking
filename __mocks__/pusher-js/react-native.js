const channel = {
  bind: jest.fn(),
};

const pusherClient = {
  subscribe: jest.fn(),
  unsubscribe: jest.fn(),
  disconnect: jest.fn(),
};

pusherClient.subscribe.mockImplementation(() => channel);

const MockPusher = jest.fn();
MockPusher.mockImplementation(() => pusherClient);

export default MockPusher;
