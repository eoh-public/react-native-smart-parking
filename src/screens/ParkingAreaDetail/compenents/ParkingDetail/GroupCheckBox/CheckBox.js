import React, { memo, useCallback } from 'react';
import { TouchableOpacity, StyleSheet, View, Image } from 'react-native';

import { Colors } from '../../../../../configs';
import Text from '../../../../../commons/Text';
import { RadioCircle } from '../../../../../commons';

const CheckBox = memo(
  ({
    select,
    title,
    index,
    onSelect,
    source,
    description,
    value,
    noLine,
    disabled,
  }) => {
    const onPress = useCallback(() => {
      onSelect && onSelect(index);
    }, [index, onSelect]);
    const opacity = disabled ? 0.5 : 1;
    return (
      <TouchableOpacity
        style={[styles.container, { opacity: opacity }]}
        onPress={onPress}
        activeOpacity={0.4}
        disabled={disabled}
      >
        <RadioCircle active={select} style={styles.svgCheck} />
        <View style={styles.contentText}>
          <Text type="H4" color={Colors.Gray9} style={styles.text}>
            {title}
            {!!description && (
              <Text type="Body" color={Colors.Gray8}>
                {`${'\n\n'}${description}`}
              </Text>
            )}
          </Text>
        </View>
        <Image source={{ uri: source }} style={styles.image} />
        {!noLine && <View style={styles.line} />}
      </TouchableOpacity>
    );
  }
);
export default CheckBox;
const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    paddingVertical: 16,
    overflow: 'hidden',
  },
  svgCheck: {
    marginTop: 2,
  },
  text: {
    flex: 1,
    marginLeft: 8,
  },
  line: {
    height: 1,
    position: 'absolute',
    width: '100%',
    backgroundColor: Colors.Gray4,
    bottom: 0,
    marginLeft: 24,
  },
  contentText: {
    flex: 1,
  },
  image: {
    width: 30,
    height: 30,
  },
});
