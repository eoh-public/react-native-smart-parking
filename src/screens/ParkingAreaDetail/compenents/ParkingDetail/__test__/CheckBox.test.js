import CheckBox from 'components/GroupCheckBox/CheckBox.js';
import React from 'react';
import { TouchableOpacity } from 'react-native';

import renderer, { act } from 'react-test-renderer';

describe('Test CheckBox', () => {
  let wrapper;
  test('create CheckBox', () => {
    const mockFunc = jest.fn();
    act(() => {
      wrapper = renderer.create(
        <CheckBox
          key={'1'}
          title={'ABC'}
          index={1}
          onSelect={mockFunc}
          select={'select'}
          source={'source'}
          description={'description'}
        />
      );
    });

    const testInstance = wrapper.root;
    const button = testInstance.findByType(TouchableOpacity);
    renderer.act(() => {
      button.props.onPress();
    });
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
