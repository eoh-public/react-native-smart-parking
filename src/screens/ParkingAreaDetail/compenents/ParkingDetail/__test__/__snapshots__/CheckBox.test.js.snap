// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CheckBox create CheckBox 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexDirection": "row",
      "opacity": 1,
      "overflow": "hidden",
      "paddingVertical": 12,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#00979D",
        "borderRadius": 8,
        "borderWidth": 1,
        "height": 16,
        "justifyContent": "center",
        "width": 16,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#00979D",
          "borderRadius": 4,
          "height": 8,
          "width": 8,
        }
      }
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#262626",
          "fontFamily": "SFProDisplay-Regular",
          "fontSize": 16,
          "lineHeight": 16,
          "textAlign": "left",
          "textDecorationLine": "none",
        },
        Object {
          "flex": 1,
          "marginLeft": 8,
        },
      ]
    }
  >
    ABC
    <Text
      style={
        Array [
          Object {
            "color": "#595959",
            "fontFamily": "SFProDisplay-Regular",
            "fontSize": 14,
            "lineHeight": 14,
            "textAlign": "left",
            "textDecorationLine": "none",
          },
          Object {},
        ]
      }
    >
      {
}description
    </Text>
  </Text>
  <View
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "height": 24,
          "width": 24,
        },
      ]
    }
  >
    <FastImageView
      resizeMode="cover"
      source={null}
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#E8E8E8",
        "bottom": 0,
        "height": 1,
        "marginLeft": 24,
        "position": "absolute",
        "width": "100%",
      }
    }
  />
</View>
`;
